### PR TITLE
Fix https://sft.its.cern.ch/jira/browse/ROOT-9253

### PR DESCRIPTION
**Pull request by pcanal**
Narrow the lock scope and its rate of use.

### CODE REVIEW

**Comment by dpiparo** [comment]
@pcanal : this looks good to me and I am in favour of merging soon.

**Comment by pcanal** [comment]
failure in unrelated.

File: tree/tree/src/TBranchElement.cxx [hunk 1]
```diff
@@ -5239,6 +5248,7 @@ void TBranchElement::SetAddress(void* addr)
    // We do this only once because it depends only on
    // the type of our object, not on its address.
    if (!fInitOffsets) {
```
Comment (Dr15Jones) [inline tree/tree/src/TBranchElement.cxx]: Is this check thread-safe?

Comment (pcanal) [reply tree/tree/src/TBranchElement.cxx]: No, but it does not need to.  Only one thread should call SetAddress on the same branch object at a time.

**Pull request merged**